mlflow/pipelines/steps/train.py

```diff
@@ -124,135 +124,181 @@ def _run(self, output_directory):
             MLFLOW_PIPELINE_STEP_NAME: run_args.get("step", ""),
         }
 
-        if self.step_config["tuning_enabled"]:
-            # gate all HP tuning code within this condition
-            tuning_params = self.step_config["tuning"]
-            # import hyperopt or throw error
-            try:
-                from hyperopt import hp, fmin
-            except ModuleNotFoundError:
-                raise MlflowException(
-                    "Hyperopt not installed and is required if tuning is enabled",
-                    error_code=BAD_REQUEST,
-                )
-
-            # wrap training in objective fn
-            def objective(args):  # pylint: disable=unused-argument
-                return None
-
-            # construct hp search space from yaml
-            search_space = {}
-            params = tuning_params["parameters"]
-            for param_name, param_details in params.items():
-                if "values" in param_details:
-                    search_space[param_name] = hp.choice(param_name, **param_details)
-                elif "distribution" in param_details:
-                    hp_tuning_fn = getattr(hp, param_details["distribution"])
-                    param_details_to_pass = param_details.copy()
-                    param_details_to_pass.pop("distribution")
-                    search_space[param_name] = hp_tuning_fn(param_name, **param_details_to_pass)
-                else:
+        mlflow.autolog(log_models=False)
+        with mlflow.start_run(tags=tags) as run:
+            estimator_hardcoded_params = self.step_config["estimator_params"]
+            if self.step_config["tuning_enabled"]:
```

Comment:
As discussed offline, can we factor out the tuning logic into multiple helper methods that are invoked from within this branch and deduplicate some of the code that might be shared with the non-tuning branch?

Reply:
@prithvikannan Can we move the tuning logic into a separate method so that we can just call

```
if self.step_config["tuning_enabled"]:
    estimator = tune_and_get_best_estimator(...)
else:
    estimator = estimator_fn(estimator_hardcoded_params)
```

?